objetos/funcionarios/Funcionario.java

```diff
@@ -65,6 +70,28 @@ public boolean setNivelCargo(String nivelCargo) {
         }
     }
 
+    //Promover - apenas salário
+    public void promover(double valorAdicionalSalario) {
+        this.valorAdicionalSalario = valorAdicionalSalario;
+        this.setSalario(this.salario + valorAdicionalSalario);
+    }
+
+    //Promover - Cargo com aumento salarial
+    public void promover(String nivelCargo, double valorAdicionalSalario) {
```

Comment:
No método promover você está passando como parâmetro o nível do cargo. 
Nossa sugestão foi não passar mais o parâmetro e deixar essa alteração de uma forma lógica.
Ou seja, 
SE this.nivelCargo = INICIANTE ele mudará para EXPERIENTE
SE this.nivelCargo = EXPERIENTE ele mudará para AVANCADO
SE this.nivelCargo = AVANCADO ele dá erro pq não tem nível maior!

Sendo assim ele receberia só por parâmetro o novo salário que subistituirá o salário antigo.

Reply:
feito